File: Bindings~/java/src/main/java/com/alttester/Commands/UnityCommand/AltGetTimeScale.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.UnityCommand;

import com.alttester.AltMessage;
import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltGetTimeScale extends AltBaseCommand {

  public AltGetTimeScale(IMessageHandler messageHandler) {
    super(messageHandler);
  }

  public float Execute() {
    AltMessage altMessage = new AltMessage();
    altMessage.setCommandName("getTimeScale");
    SendCommand(altMessage);
    return recvall(altMessage, Float.class);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/UnityCommands/AltGetTimeScale.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

using Newtonsoft.Json;

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltGetTimeScale : AltBaseCommand
    {
        AltGetTimeScaleParams cmdParams;
        public AltGetTimeScale(IDriverCommunication commHandler) : base(commHandler)
        {
            cmdParams = new AltGetTimeScaleParams();
        }
        public float Execute()
        {
            CommHandler.Send(cmdParams);
            return CommHandler.Recvall<float>(cmdParams);
        }
    }
}
